app/code/Magento/WishlistGraphQl/Model/Resolver/WishlistRemoveItemResolver.php

```diff
@@ -0,0 +1,164 @@
+use Magento\Wishlist\Model\ResourceModel\Item\CollectionFactory as WishlistItemCollectionFactory;
+use Magento\Wishlist\Model\ResourceModel\Wishlist as WishlistResourceModel;
+use Magento\Wishlist\Model\ResourceModel\Item as WishlistItemResourceModel;
+use Magento\Wishlist\Model\Wishlist;
+use Magento\Wishlist\Model\WishlistFactory;
+use Magento\WishlistGraphQl\Model\GetWishlistByAttribute;
+
+/**
+ * Remove wishlist items according to the GraphQL schema
+ */
+class WishlistRemoveItemResolver implements ResolverInterface
+{
+    /**
+     * @var WishlistResourceModel
+     */
+    private $wishlistResource;
+
+    /**
+     * @var WishlistItemResourceModel
+     */
+    private $wishlistItemResource;
+
+    /**
+     * @var WishlistFactory
+     */
+    private $wishlistFactory;
+
+    /**
+     * @var WishlistItemCollectionFactory
+     */
+    private $wishlistItemCollectionFactory;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var GetWishlistByAttribute
+     */
+    private $getWishlistByAttribute;
+
+    /**
+     * @param WishlistResourceModel $wishlistResource
+     * @param WishlistItemResourceModel $wishlistItemResource
+     * @param WishlistFactory $wishlistFactory
+     * @param WishlistItemCollectionFactory $wishlistItemCollectionFactory
+     * @param StoreManagerInterface $storeManager
+     * @param GetWishlistByAttribute $getWishlistByAttribute
+     */
+    public function __construct(
+        WishlistResourceModel $wishlistResource,
+        WishlistItemResourceModel $wishlistItemResource,
+        WishlistFactory $wishlistFactory,
+        WishlistItemCollectionFactory $wishlistItemCollectionFactory,
+        StoreManagerInterface $storeManager,
+        GetWishlistByAttribute $getWishlistByAttribute
+    ) {
+        $this->wishlistResource = $wishlistResource;
+        $this->wishlistItemResource = $wishlistItemResource;
+        $this->wishlistFactory = $wishlistFactory;
+        $this->wishlistItemCollectionFactory = $wishlistItemCollectionFactory;
+        $this->storeManager = $storeManager;
+        $this->getWishlistByAttribute = $getWishlistByAttribute;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @throws GraphQlAuthorizationException
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $customerId = $context->getUserId();
+
+        if (!$customerId && $customerId === 0) {
+            throw new GraphQlAuthorizationException(__('The current user cannot perform operations on wishlist'));
+        }
+
+        $wishlist = $this->getWishlistByAttribute->execute('wishlist_id', $args['input']['wishlist_id']);
+
+        if ((int) $wishlist->getCustomerId() !== $customerId) {
+            throw new GraphQlAuthorizationException(__('The current user doesn\'t have a wishlist with the provided id'));
```

Comment:
Let's do not expose additional info. Message should be the same `throw new GraphQlAuthorizationException(__('The current customer isn\'t authorized.'));`